### ISSUE TITLE
Allow u.quantity_input to accept None as default kwarg? 

### ISSUE THREAD

**Issue by Cadair** [astropy/astropy#3734]
Currently this:

``` python
@u.quantity_input(angle=u.deg):
def rotate(angle=None):
    pass
```

Will error. Do we want to allow this as a valid pattern?

**Comment by astrofrog** [comment]
Yes, I think that would make sense. Maybe it could somehow be an option in the decorator in case people want to enable/disable it?

**Comment by Cadair** [comment]
@astrofrog assign me. You had to ask for the option to turn it off didn't you!

**Comment by astrofrog** [comment]
:wink:

**Comment by astrofrog** [comment]
Having said that, I think it would be fine to have a single option that applies to all arguments, i.e. ``allow_none=True`` (the default). Don't think we need more granular control.

**Comment by Cadair** [comment]
@astrofrog that was all you were getting anyway. :p 

More seriously, I agree.

**Comment by embray** [comment]
Does it really even need to be an option in the decorator?  If the function defines `None` as the *default* value for some arguments, then I think the decorator should allow it (and otherwise not allow None on other arguments).

**Comment by mhvk** [comment]
@Cadair - looking over old units issues, I came across this one. My own sense is to agree with @embray, that this doesn't need an argument, but just should follow what the author does.

**Comment by adrn** [comment]
Missed this issue when I had my own similar one, which was closed by the (merged) implementation in #5653!
